### ISSUE TITLE
Here is the code adapted from your project that results in false positives on human voices. Please advise

### ISSUE THREAD

**Issue by KabaTubare** [facebookresearch/audioseal#17]
Automatically generated by Colaboratory.

Original file is located at
    https://colab.research.google.com/drive/1LuUOwwVg5s8jcZ0n5lddLR3jBr0LIKUu
"""

!pip install torchaudio
!pip install matplotlib
!pip install audioseal # Ensure this matches the actual package name for AudioSeal

import io
import torch
import torchaudio
import matplotlib.pyplot as plt
from IPython.display import Audio as IPythonAudio
from google.colab import files

try:
    from audioseal import AudioSeal
    audioseal_available = True
    print("AudioSeal is available for watermark detection.")
except ImportError:
    audioseal_available = False
    print("AudioSeal is not available. Please ensure it's correctly installed.")

def upload_sample_audio():
    """Use Google Colab's file upload dialog to upload and load a sample audio file."""
    uploaded = files.upload()  # Display the file upload dialog
    if not uploaded:
        print("No file uploaded.")
        return None, None
    file_name = next(iter(uploaded))
    waveform, sample_rate = torchaudio.load(io.BytesIO(uploaded[file_name]))
    return waveform, sample_rate

def plot_waveform_and_specgram(waveform, sample_rate, title="Audio"):
    """Plot the waveform and spectrogram of the audio."""
    fig, axs = plt.subplots(2, figsize=(20, 10))
    axs[0].plot(waveform.t().numpy())
    axs[0].set_title(f'{title} Waveform')
    axs[0].set_ylabel('Amplitude')
    specgram = torchaudio.transforms.Spectrogram()(waveform)
    axs[1].imshow(specgram.log2()[0,:,:].numpy(), cmap='viridis', aspect='auto')
    axs[1].set_title(f'{title} Spectrogram')
    axs[1].set_ylabel('Frequency bins')
    axs[1].set_xlabel('Time frames')
    plt.show()

def play_audio(waveform, sample_rate):
    """Play the uploaded audio."""
    display(IPythonAudio(waveform.numpy(), rate=sample_rate))

def detect_watermark(waveform, sample_rate):
    """Detect watermark in the uploaded audio using AudioSeal."""
    if audioseal_available:
        detector = AudioSeal.load_detector("audioseal_detector_16bits")
        result, message = detector.detect_watermark(waveform.unsqueeze(0), message_threshold=0.5)
        print(f"Detection result: {'Watermarked or AI-Generated audio' if result else 'Not watermarked or AI-Generated'}")
    else:
        print("Watermark detection not available. AudioSeal is not installed.")

if __name__ == "__main__":
    audio, sr = upload_sample_audio()
    if audio is not None and sr is not None:
        plot_waveform_and_specgram(audio, sr, "Uploaded Audio")
        play_audio(audio, sr)
        detect_watermark(audio, sr)
    else:
        print("Audio upload failed. Please try again.")

--------------------

"""audioseal_watermarking_and_detection_27_2_2024

Automatically generated by Colaboratory.

Original file is located at
    https://colab.research.google.com/drive/1bbCzpIRgA5pBWsx0dM3NIO3XhIglKQHu
"""

 Install necessary libraries including AudioSeal
!pip install torchaudio soundfile matplotlib
!pip install audioseal

import torch
import torchaudio
import urllib.request
import matplotlib.pyplot as plt
from IPython.display import display, Audio, HTML
import torchaudio.transforms as T
from google.colab import files

uploaded = files.upload()

filename = next(iter(uploaded))
print(f"Uploaded file: {filename}")

display(HTML("<style>.container { width:100% !important; }</style>"))

plt.rcParams["figure.figsize"] = (20, 3)

audio, sr = torchaudio.load(filename)

from IPython.display import Audio

def play_audio(waveform, sample_rate):
    display(Audio(waveform.numpy(), rate=sample_rate))

def plot_waveform(waveform, sample_rate, title="Waveform"):
    plt.figure(figsize=(20, 4))
    
(channels, samples) and select the first channel if more than one
    if waveform.dim() == 3:
        waveform = waveform[0]  # Select first batch
    if waveform.size(0) > 1:  # More than one channel
        waveform = waveform[0]  # Select first channel

    plt.plot(waveform.numpy())
    plt.title(title)
    plt.ylabel('Amplitude')
    plt.xlabel('Time (samples)')
    plt.xlim([0, waveform.size(-1)])
    plt.show()

def plot_spectrogram(waveform, sample_rate, title="Spectrogram", n_fft=400):

    if waveform.dim() == 3 and waveform.size(0) == 1:
        waveform = waveform.squeeze(0)  # Remove batch dimension if it's 1

    spectrogram_transform = T.Spectrogram(n_fft=n_fft, power=2)
    spectrogram = spectrogram_transform(waveform)

    spectrogram_db = torchaudio.transforms.AmplitudeToDB()(spectrogram)

    plt.figure(figsize=(20, 4))
  
    if spectrogram_db.dim() > 2:
        spectrogram_db = spectrogram_db.squeeze(0)
    plt.imshow(spectrogram_db.numpy(), cmap='viridis', aspect='auto')
    plt.title(title)
    plt.ylabel('Frequency bins')
    plt.xlabel('Time (frames)')
    plt.colorbar(format="%+2.0f dB")
    plt.show()

plot_waveform(audio, sr, "Original Audio Waveform")
plot_spectrogram(audio, sr, "Original Audio Spectrogram")
play_audio(audio, sr)

from audioseal import AudioSeal

model = AudioSeal.load_generator("audioseal_wm_16bits")

audios = audio.unsqueeze(0)

watermark = model.get_watermark(audios)
watermarked_audio = audios + watermark

/ tune-up rate
watermarked_audio = model(audios, alpha=1)

watermarked_audio_detached = watermarked_audio.detach()

watermarked_audio_detached = watermarked_audio.detach()

watermarked_audio_detached.dim() == 3 and watermarked_audio_detached.shape[0] == 1:
    watermarked_audio_detached = watermarked_audio_detached.squeeze(0)

plot_waveform(watermarked_audio_detached, sr, "Watermarked Audio")
plot_spectrogram(watermarked_audio_detached, sr, "Watermarked Spectrogram")
play_audio(watermarked_audio_detached, sr)  # Remove .squeeze() if you want to play stereo

secret_message = torch.randint(0, 2, (1, 16), dtype=torch.int32)
watermarked_audio = model(audios, message=secret_message, alpha=1)

detector = AudioSeal.load_detector("audioseal_detector_16bits")

result, message = detector.detect_watermark(watermarked_audio, message_threshold=0.5)
print(f"\nThis is likely a watermarked audio: {result}")

result2, message2 = detector.detect_watermark(audios, message_threshold=0.5)
print(f"This is likely an unwatermarked audio: {result2}")

from audioseal.attacks import AudioEffects as af

pink_noised_audio = af.pink_noise(watermarked_audio, noise_std=0.2)
plot_waveform_and_specgram(pink_noised_audio, sr, title="Audio with Pink Noise")
result, message = detector.detect_watermark(pink_noised_audio)
print(f"Pink noise attack result: {result}")

lowpass_filtered = af.lowpass_filter(watermarked_audio, cutoff_freq=5000, sample_rate=sr)
plot_waveform_and_specgram(lowpass_filtered, sr, title="Lowpass Filtered Audio")
result, message = detector.detect_watermark(lowpass_filtered)
print(f"Lowpass filter attack result: {result}")

--------

**Comment by pierrefdz** [comment]
Transferred to issue #14 to remove duplicates
